src/middlewared/middlewared/plugins/vm.py

```diff
@@ -1257,6 +1273,56 @@ class Config:
         datastore = 'vm.device'
         datastore_extend = 'vm.device.extend_device'
 
+    @private
+    async def nic_check(self, vm_device):
+        if not await self.middleware.call('system.is_freenas') and await self.middleware.call('failover.licensed'):
+            failover_enabled = not (await self.middleware.call('failover.config'))['disabled']
+            nics = await self.nic_capability_checks([vm_device])
+            if nics:
+                if failover_enabled:
+                    raise CallError(
+                        'Failover must be disabled before attempting to create/update VM NIC device '
+                        f'because of the following nics: {",".join(nics)}'
+                    )
+                else:
+                    for nic in nics:
+                        await self.middleware.call('interface.disable_capabilities', nic, nics[nic])
+                        try:
+                            await self.middleware.call(
+                                'failover.call_remote', 'interface.disable_capabilities', [nic, nics[nic]]
+                            )
+                        except Exception as e:
+                            self.middleware.logger.debug(
+                                f'Failed to disable capabilities for {nic} on remote node: {e}'
```

Comment:
The `remote node` was removed as accepted verbiage for TrueNAS HA. Let's use `for {nic} on standby storage controller`, please.